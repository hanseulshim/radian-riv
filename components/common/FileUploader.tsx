import React, { useRef } from 'react'

interface Props {
  handleFile: (e) => void
  selectedFile: File
  accept?: string
}

export default function FileUploader({
  handleFile,
  selectedFile,
  accept
}: Props) {
  const hiddenFileInput = useRef(null)

  const handleClick = event => {
    hiddenFileInput.current.click()
  }

  const handleChange = event => {
    const fileUploaded = event.target.files[0]
    handleFile(fileUploaded)
  }
  return (
    <div className="custom-file-uploader">
      <button className="btn btn-small" onClick={handleClick} type="button">
        Choose File...
      </button>
      <span>{selectedFile ? selectedFile.name : 'No file chosen'}</span>
      <input
        type="file"
        ref={hiddenFileInput}
        onChange={handleChange}
        style={{ display: 'none' }}
        accept={accept}
      />
    </div>
  )
}
