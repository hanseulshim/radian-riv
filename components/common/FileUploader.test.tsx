import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import FileUploader from './FileUploader'

describe('FileUploader', () => {
  let component
  beforeAll(() => {
    const props = {
      handleFile: () => {},
      selectedFile: null
    }
    component = shallow(<FileUploader {...props} />)
  })
  it('renders container', () => {
    expect(component.find('.custom-file-uploader')).toHaveLength(1)
  })
  it('has no label if no file is selected', () => {
    expect(component.find('span').exists()).toBeFalsy()
  })
  it('should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
