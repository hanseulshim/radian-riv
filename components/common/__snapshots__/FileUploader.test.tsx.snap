// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploader should match snapshot 1`] = `
<div
  className="custom-file-uploader"
>
  <button
    className="btn btn-small"
    onClick={[Function]}
    type="button"
  >
    Choose File...
  </button>
  <span>
    No file chosen
  </span>
  <input
    onChange={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
